restli-common/src/main/java/com/linkedin/restli/internal/common/URIParamUtils.java

```diff
@@ -478,6 +480,30 @@ public static URI replaceQueryParam(URI uri, String queryParam, DataComplex valu
     return builder.build();
   }
 
+  private static void convertRawProjectionFieldsToMaskTree(DataMap queryParams)
```

Comment:
Is this extra logic necessary? Isn't the parsed mask tree available in the response builder?

Reply:
yes, you are right,  the mastrees are generated during the class ResourceContextImpl constructor, so to avoid this extra step, I just need a reference from ResourceContextImpl instance